src/networking.c

```diff
@@ -4256,42 +4257,44 @@ void killIOThreads(void) {
     }
 }
 
-void startThreadedIO(void) {
-    serverAssert(server.io_threads_active == 0);
-    for (int j = 1; j < server.io_threads_num; j++) pthread_mutex_unlock(&io_threads_mutex[j]);
-    server.io_threads_active = 1;
-}
```

Comment:
These changes will just cause merge conflicts for the Async IO threads feature. It's better that we don't merge it now.

Reply:
I saw the Async IO threads remove the `startThreadIO` and use the dynamic threads number too, this is similar with this patch. The difference is that Async IO threads keep the `threshold` based on the real time events count, this patch use  EMA to make a more consistent latency when requests spikes/troughs. 

Maybe we could merge this firstly and then rebase Async IO threads on this? If needed, I could help the rebase work.

Reply:
@uriyage WDYT?